### DISCUSSION TITLE
copy or move a range of cells to new co-ordinates?

### DISCUSSION THREAD

**Discussion by looksystems** [PHPOffice/PhpSpreadsheet#2941] [General]
Does anyone know if it is possible to copy or move a range of cells to new co-ordinates, updating cell references and maintaining styles accordingly?

Unless I missed something, I can't see anything the discussions, issues, online docs or any related methods in the Worksheet or Collection\Cells api. 

Any pointers would be most gratefully received, thank you.

**Comment by MarkBaker** [comment]
There's no specific mechanism that we provide for copying or moving cells

You could probably do something using the ReferenceHelper; that's the class that handles inserting/deleting rows/columns while adjusting all their references

**Reply by looksystems** [reply]
So change the copied cell value using this method, is that right?

```
    /**
     * Update references within formulas.
     *
     * @param string $formula Formula to update
     * @param string $beforeCellAddress Insert before this one
     * @param int $numberOfColumns Number of columns to insert
     * @param int $numberOfRows Number of rows to insert
     * @param string $worksheetName Worksheet name/title
     *
     * @return string Updated formula
     */
    public function updateFormulaReferences(
        $formula = '',
        $beforeCellAddress = 'A1',
        $numberOfColumns = 0,
        $numberOfRows = 0,
        $worksheetName = '',
        bool $includeAbsoluteReferences = false
    ) {
```

**Reply by MarkBaker** [reply]
Something using that should work; depends if you also need to change any DataValidation, Hyperlinks, etc which have separate methods in the ReferenceHelper

**Reply by looksystems** [reply]
My use case really only cares about the formula, itself and the format code.

Oh, just realised, that's great for the "copy" operation - but - for a "move" operation, I guess I also have to iterate across the whole spreadsheet and fix any references to the moved cells too.

Anyway, thank for you help(!) and amazing super quick response.
